### PR TITLE
RFC: Decision log for "Portal Host Detection"

### PR DESCRIPTION
**Pull request by mjbrisebois**
Please add comments to this pull request.

Use the suggestion feature to request changes or to submit additional options/pros/cons.

### CODE REVIEW

**Comment by mjbrisebois** [comment]
My opinion for the best option is currently the "Ping/Pong Vector Call"

Reasons

- @thedavidmeister pointed out that the vector call is already a concession due to lack of async
  support in wasm.  Expanding the feature is spending more effort in a direction that will most
  likely be undesirable in the future.
  - https://chat.holochain.org/holo/pl/chy83satctn4bcf8h89eacfi4o
- We can communicate to the user how long the wait will take so they are not confused during the wait

File: rfcs/DL-2023-02-27-01-portal-host-detection.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+When an Agent is completely offline, it will take the full Holochain timeout to get a response.
+Holochain currently supports a vector of calls so that a list of potential hosts can be queried all
+at once.  This still results in the full wait for the timeout if any of the hosts are offline.
+
+This is because QUIC does not differentiate between the connection timeout and the request timeout.
+So this problem should go away with WebRTC.
+
+
+## Options
```
Comment (guillemcordoba) [inline rfcs/DL-2023-02-27-01-portal-host-detection.md:22]: Another option would be similar to Ping/Pong vector call, but driving all the calls from the UI side. App store would implement a zome function that given an agent public key of a host, it's tries to call_remote to that host, and another zome function that returns the list of hosts available to the UI. Then, the UI would initiate one async call per host to try to connect to each of them, and cancel all of them as soon as some answer gets returned.

**Comment by mjbrisebois** [comment]
> My opinion for the best option is currently the "Ping/Pong Vector Call"
> 
> Reasons
> 
> * @thedavidmeister pointed out that the vector call is already a concession due to lack of async
>   support in wasm.  Expanding the feature is spending more effort in a direction that will most
>   likely be undesirable in the future.
>   
>   * https://chat.holochain.org/holo/pl/chy83satctn4bcf8h89eacfi4o
> * We can communicate to the user how long the wait will take so they are not confused during the wait

I like the new option proposed by @guillemcordoba which appears to have no cons

**Comment by thedavidmeister** [comment]
yeah UI seems like a good way forward if you can do it

**Pull request merged**